<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Config</class>
 <widget class="QDialog" name="Config">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>509</width>
    <height>699</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Configuration</string>
  </property>
  <widget class="QWidget" name="formLayoutWidget">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>30</y>
     <width>441</width>
     <height>301</height>
    </rect>
   </property>
   <layout class="QFormLayout" name="formLayoutRC">
    <item row="0" column="0">
     <widget class="QLabel" name="labelModelId">
      <property name="text">
       <string>Model ID</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLineEdit" name="LeModelId">
      <property name="toolTip">
       <string>Rotctld Rotor Model ID</string>
      </property>
      <property name="inputMask">
       <string>009</string>
      </property>
      <property name="placeholderText">
       <string>603</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="labelSerialDev">
      <property name="text">
       <string>Device (Serial)</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="LeDevSerial">
      <property name="toolTip">
       <string>Rotctld Serial Device Path</string>
      </property>
      <property name="maxLength">
       <number>512</number>
      </property>
      <property name="placeholderText">
       <string>/dev/ttyACM0</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="labelSpeedSerial">
      <property name="text">
       <string>Serial Speed</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QComboBox" name="CbSpeedSerial">
      <property name="toolTip">
       <string>Serial Baud Rate</string>
      </property>
      <property name="currentText">
       <string>9600</string>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <item>
       <property name="text">
        <string>1200</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>2400</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>4800</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>9600</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>19200</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>38400</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>57600</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>115200</string>
       </property>
      </item>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="labelRcExePath">
      <property name="text">
       <string>Path to Rotctld</string>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QLineEdit" name="LeRcExePath">
      <property name="toolTip">
       <string>Path to Rotctld Executable</string>
      </property>
      <property name="maxLength">
       <number>1024</number>
      </property>
      <property name="placeholderText">
       <string>/usr/bin/rotctld</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="labelRcStartup">
      <property name="text">
       <string>Startup Rotctld</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QCheckBox" name="ChboxRcStartup">
      <property name="text">
       <string>on Program Start</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="labelRcPoll">
      <property name="text">
       <string>Poll Rate (ms)</string>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QLineEdit" name="LeRcPollRate">
      <property name="toolTip">
       <string>Update Rate for Rotctl</string>
      </property>
      <property name="inputMethodHints">
       <set>Qt::ImhNone</set>
      </property>
      <property name="inputMask">
       <string/>
      </property>
      <property name="maxLength">
       <number>5</number>
      </property>
      <property name="placeholderText">
       <string>1000</string>
      </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QLabel" name="labelRcPort">
      <property name="text">
       <string>Rotctld Port</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1">
     <widget class="QLineEdit" name="LeRcPort">
      <property name="toolTip">
       <string>Rotctld TCP Port Number (4533 is default)</string>
      </property>
      <property name="inputMask">
       <string/>
      </property>
      <property name="maxLength">
       <number>1024</number>
      </property>
      <property name="placeholderText">
       <string>localhost:4533</string>
      </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QLabel" name="labelFilename">
      <property name="text">
       <string>Config File</string>
      </property>
     </widget>
    </item>
    <item row="7" column="1">
     <widget class="QLineEdit" name="LeFilename">
      <property name="toolTip">
       <string>Config File Path</string>
      </property>
      <property name="maxLength">
       <number>1024</number>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QLabel" name="labelMainTitle">
      <property name="text">
       <string>Main Title</string>
      </property>
     </widget>
    </item>
    <item row="8" column="1">
     <widget class="QLineEdit" name="LeMainTitle">
      <property name="toolTip">
       <string>Application Display Title</string>
      </property>
      <property name="maxLength">
       <number>50</number>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPlainTextEdit" name="PteRcLog">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>480</y>
     <width>441</width>
     <height>191</height>
    </rect>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>350</y>
     <width>441</width>
     <height>91</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="PbRcTest">
        <property name="toolTip">
         <string>Test if Rotctld is running</string>
        </property>
        <property name="text">
         <string>Test Config</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="PbRcStart">
        <property name="toolTip">
         <string>Startup Rotctld daemon</string>
        </property>
        <property name="text">
         <string>Start Rotctld</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="PbRcStop">
        <property name="toolTip">
         <string>Stop running Rotctld</string>
        </property>
        <property name="text">
         <string>Stop Rotctld</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="PbSave">
        <property name="toolTip">
         <string>Save Current Settings to Config File</string>
        </property>
        <property name="text">
         <string>Save</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="PbOpen">
        <property name="toolTip">
         <string>Open Config file</string>
        </property>
        <property name="text">
         <string>Open</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDialogButtonBox" name="buttonBox">
        <property name="toolTip">
         <string/>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="standardButtons">
         <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
        </property>
        <property name="centerButtons">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QLabel" name="labelLog">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>460</y>
     <width>111</width>
     <height>17</height>
    </rect>
   </property>
   <property name="text">
    <string>Log Messages</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Config</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Config</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LeModelId</sender>
   <signal>editingFinished()</signal>
   <receiver>Config</receiver>
   <slot>LeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>306</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LeDevSerial</sender>
   <signal>editingFinished()</signal>
   <receiver>Config</receiver>
   <slot>LeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>380</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LeRcExePath</sender>
   <signal>editingFinished()</signal>
   <receiver>Config</receiver>
   <slot>LeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>323</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LeRcPollRate</sender>
   <signal>editingFinished()</signal>
   <receiver>Config</receiver>
   <slot>LeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>220</x>
     <y>204</y>
    </hint>
    <hint type="destinationlabel">
     <x>220</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LeRcPort</sender>
   <signal>editingFinished()</signal>
   <receiver>Config</receiver>
   <slot>LeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>246</x>
     <y>237</y>
    </hint>
    <hint type="destinationlabel">
     <x>251</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PbSave</sender>
   <signal>clicked()</signal>
   <receiver>Config</receiver>
   <slot>PbSave_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>114</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PbOpen</sender>
   <signal>clicked()</signal>
   <receiver>Config</receiver>
   <slot>PbOpen_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>11</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ChboxRcStartup</sender>
   <signal>stateChanged(int)</signal>
   <receiver>Config</receiver>
   <slot>ChboxRcStartup_StateChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>264</x>
     <y>172</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LeFilename</sender>
   <signal>editingFinished()</signal>
   <receiver>Config</receiver>
   <slot>LeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>314</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>316</x>
     <y>296</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LeMainTitle</sender>
   <signal>editingFinished()</signal>
   <receiver>Config</receiver>
   <slot>LeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>274</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PbRcTest</sender>
   <signal>clicked()</signal>
   <receiver>Config</receiver>
   <slot>PbRcTest_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>373</y>
    </hint>
    <hint type="destinationlabel">
     <x>61</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PbRcStart</sender>
   <signal>clicked()</signal>
   <receiver>Config</receiver>
   <slot>PbRcStart_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>198</x>
     <y>368</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PbRcStop</sender>
   <signal>clicked()</signal>
   <receiver>Config</receiver>
   <slot>PbRcStop_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>371</y>
    </hint>
    <hint type="destinationlabel">
     <x>427</x>
     <y>338</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>LeChanged()</slot>
  <slot>PbSave_Clicked()</slot>
  <slot>PbOpen_Clicked()</slot>
  <slot>ChboxRcStartup_StateChange(int)</slot>
  <slot>PbRcTest_Clicked()</slot>
  <slot>PbRcStart_Clicked()</slot>
  <slot>PbRcStop_Clicked()</slot>
 </slots>
</ui>
